### DISCUSSION TITLE
Color modes not working

### DISCUSSION THREAD

**Discussion by simonvargasz** [twbs/bootstrap#38655] [Q&A] [answered]
Hi, im creating a laravel app and using bootstrap for front-end, im using vite for compiling. But for some reason the data-bs-theme="dark", or any custom color mode are not working. 

This is how my navbar looks like:
![image](https://github.com/twbs/bootstrap/assets/107783058/8c9051ab-119d-449c-ab3d-3c099aa3e7bf)

This is my navbar in html:
`<nav class="navbar navbar-expand-lg rounded-3 mt-4 p-0 " data-bs-theme="dark" >`

i even tried using text-white but it doesn't work
please if someone could help me with this...

**Comment by palonza** [comment]
Hi @simonvargasz 

super fast response without seeing the code:

`<nav class="navbar navbar-expand-lg rounded-3 mt-4 p-0 " data-bs-theme="dark" style="color: white">`

force:
`<nav class="navbar navbar-expand-lg rounded-3 mt-4 p-0 " data-bs-theme="dark" style="color: white !important">`

tell me if you can do it! ;)

**Reply by simonvargasz** [reply]
Thanks for the fast response! I have tried that solution, it still does not work

**Reply by palonza** [reply]
do you can send me entire element and sub-elements?

**Reply by simonvargasz** [reply]
<nav class="navbar navbar-expand-lg rounded-3 mt-4 p-0 " data-bs-theme="dark" style="color: white !important">
            <div class="container-fluid p-0">
              <a class="navbar-brand rounded-3 me-2" style="background: rgb(31,64,94)" href="#">
                <div class="container rounded-3" style="background: rgb(31,64,94)">
                <i class="fa-solid fa-house"></i>
                </div>
              </a>
              <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
              </button>
              <div class="collapse navbar-collapse p-0" id="navbarSupportedContent">

                <div class="col-10">
                  <div class="container rounded-3 " style="background: rgb(31,64,94);
                  background: linear-gradient(90deg, rgba(31,64,94,1) 0%, rgba(76,34,91,1) 100%);">
                    <ul class="navbar-nav me-auto mb-2 mb-lg-0">
                      <li class="nav-item">
                        <div>
                          <a class="nav-link active" aria-current="page" href="#"><i class="fa-regular fa-comment"></i> Propuestas</a>
                        </div>
                      </li>
                      <li class="nav-item">
                        <a class="nav-link" href="#"><i class="fa-regular fa-square-plus"></i> Crear</a>
                      </li>
                      <li class="nav-item">
                        <a class="nav-link"><i class="fa-regular fa-clock"></i> Estado</a>
                      </li>
                    </ul>
                  </div>
                </div>
                  
                  <div class="col-2">
                    <ul class="navbar-nav me-auto mb-2 mb-lg-0">
                      <div class="container rounded-3 ms-2" style="background: rgb(76, 34, 91);">
                        <li class="nav-item text-center">
                          <a class="nav-link" href=""><i class="fa-solid fa-user"></i> Gestion</a>
                        </li>
                      </div>
                    </ul>
                  </div>

              </div>
            </div>
          </nav>

**Reply by palonza** [reply, marked as answer]
Great! @simonvargasz if you put style="color: white" in each anchor element, then, working fine... after test, you can add a custom class. Try this for example:
```html
<a className="nav-link active" aria-current="page" href="#" style="color: white;" ><i className="fa-regular fa-comment"></i> Propuestas</a>
```
if work, then add a custom class. tell me now if working to you... ;)

**Reply by simonvargasz** [reply]
alright! that worked! but i would like to know why this happens, couse if i import bootstrap to the html with the link that the docs of bootstrap page provides, this doesn't happens, this started when i compiled bootstrap and started using sass a sass file

**Reply by palonza** [reply]
Ok... the problem that you mention is called as "priority selector"... do you must review all your code like as: import CSS files order, style inheritance,etc. Please if my answer helped you, mark it as solved and help me. Thank you. 

P.S.: here more information about [this](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Cascade_and_inheritance)
